cla_public/apps/checker/forms.py

```diff
@@ -28,11 +29,57 @@
 from cla_public.apps.checker.utils import nass, passported, money_intervals_except, money_intervals
 from cla_public.apps.checker.validators import AtLeastOne, IgnoreIf, \
     FieldValue, MoneyIntervalAmountRequired, FieldValueOrNone
+from cla_public.libs.utils import recursive_dict_update
 
 
 log = logging.getLogger(__name__)
 
 
+def set_zero_values(form):
+    """Set values on a form to zero"""
+    def set_zero(field):
+        if isinstance(field, MoneyField) or isinstance(field, IntegerField):
+            field.data = 0
+        elif isinstance(field, MoneyIntervalField):
+            field.form.per_interval_value.data = 0
+            field.form.interval_period.data = 'per_month'
+        elif isinstance(field, FormField):
+            field.form = set_zero_values(field.form)
+        elif isinstance(field, FieldList):
+            field.entries = []
+    map(set_zero, form._fields.itervalues())
+    return form
```

Comment:
Hm, lots of `isinstance`, which is a bad smell. But I can only think of one other way to do this (adding a `set_zero` method to each Field class), which would be more verbose.